### PR TITLE
SNOW-1778664 Update snowflake-jdbc to 3.20.0

### LINKED ISSUE #881: Snyk: snowflake-ingest-java net.snowflake:snowflake-jdbc 3.18.0 | Snyk ID - SNYK-JAVA-NETSNOWFLAKE-8310506
Title: Snyk: snowflake-ingest-java net.snowflake:snowflake-jdbc 3.18.0
  Additional information on Snyk can be found here: https://snyk.io/org/snowflakedb-sca-scanning-public-repo/project/0a0b0a8c-113f-4e60-98d3-357b27f5338d
  Repo: snowflake-ingest-java
  CVE: CVE-2024-43382
  Package Type: java
  Package Name: net.snowflake:snowflake-jdbc
  Package Version: 3.18.0
  Snyk ID: SNYK-JAVA-NETSNOWFLAKE-8310506
  Vulnerability URL: http://security.snyk.io/vuln/SNYK-JAVA-NETSNOWFLAKE-8310506
  Severity: high
  Introduced Date: 2024-10-31
  Projects with Vulnerability: snowflakedb/snowflake-ingest-java:e2e-jar-test/core/pom.xml
  Target File: e2e-jar-test/core/pom.xml
  JIRA Ticket: https://snowflakecomputing.atlassian.net/browse/SNOW-1778664

### LINKED ISSUE #882: Snyk: snowflake-ingest-java net.snowflake:snowflake-jdbc 3.18.0 | Snyk ID - SNYK-JAVA-NETSNOWFLAKE-8310506
Title: Snyk: snowflake-ingest-java net.snowflake:snowflake-jdbc 3.18.0
  Additional information on Snyk can be found here: https://snyk.io/org/snowflakedb-sca-scanning-public-repo/project/34b0453e-1d9a-450b-9957-893ab6eaede1
  Repo: snowflake-ingest-java
  CVE: CVE-2024-43382
  Package Type: java
  Package Name: net.snowflake:snowflake-jdbc
  Package Version: 3.18.0
  Snyk ID: SNYK-JAVA-NETSNOWFLAKE-8310506
  Vulnerability URL: http://security.snyk.io/vuln/SNYK-JAVA-NETSNOWFLAKE-8310506
  Severity: high
  Introduced Date: 2024-10-31
  Projects with Vulnerability: snowflakedb/snowflake-ingest-java:e2e-jar-test/standard/pom.xml
  Target File: e2e-jar-test/standard/pom.xml
  JIRA Ticket: https://snowflakecomputing.atlassian.net/browse/SNOW-1778664

### PR DESCRIPTION
**Pull request by sfc-gh-lsembera**
Upgrade `snowflake-jdbc` to from `3.18.0` to `3.20.0`. The previous version contains a vulnerability reported by Snyk.

Fix #881 
Fix #882

### CODE REVIEW

**Review by sfc-gh-tzhang — APPROVED** [review]
Please verify it with long running test before merging, thanks!

**Review by sfc-gh-xhuang — APPROVED** [review]
thanks for taking care of this!
It resolves CVE-2024-43382

I also enabled long running tests for you. You can add the @ignore back in after confirming they ran successfully

File: pom.xml [hunk 1]
```diff
@@ -888,7 +888,18 @@
                   </ignoreClasses>
                   <findAllDuplicates>true</findAllDuplicates>
                   <ignoreWhenIdentical>true</ignoreWhenIdentical>
+                  <dependencies>
+                    <dependency>
+                      <groupId>net.snowflake</groupId>
+                      <artifactId>snowflake-jdbc</artifactId>
+                      <ignoreClasses>
```
Comment (sfc-gh-xhuang) [inline pom.xml:895]: do you know if jdbc is pulling in protobuf too? can we ignore that?
https://github.com/snowflakedb/snowflake-ingest-java/issues/876

**Comment by sfc-gh-xhuang** [comment]
Long running tests are successful, can add the ignore back in

File: pom.xml (hunk 1 above)
Comment (sfc-gh-lsembera) [reply pom.xml:895]: This ignore is just for the `banDuplicateClasses` enforcer check, it does not exclude any classes from the JAR. Until we move to `snowflake-jdbc-thin`, we have no way of excluding classes from the JDBC driver.

**Pull request merged**